### PR TITLE
feat: Integrate GitHub Copilot Rubber Duck cross-model review (CE)

### PR DESCRIPTION
**Pull request by Jenp-AICraftWorks**
## Summary

Adds GitHub Copilot **Rubber Duck** cross-model second-opinion review support to AgentCraftworks CE.

Rubber Duck uses a different AI family from the primary orchestrator (e.g., Claude → GPT-5.x) to catch plan flaws, edge cases, and test gaps at three key checkpoints before they become execution errors. This PR brings the CE community up to date with the core platform integration.

---

### Squad Opt-Out Template
- **`.squad/rubber-duck.yml`** — Drop-in squad-level config for enabling/disabling Rubber Duck and tuning checkpoints and rate limits

### Customer Guidance
- **`docs/RUBBER_DUCK_INTEGRATION.md`** — Open-source-friendly customer guide covering:
  - How Rubber Duck works in peer-to-peer and squad patterns
  - Default behavior by engagement level
  - Opt-out instructions (repo-level, squad-level, per-task)
  - Rate Governor considerations
  - FAQ

---

## Default Behavior

| Engagement Level | Rubber Duck |
|-----------------|-------------|
| 1–2 (Observer, Advisor) | ❌ OFF |
| 3+ (Peer Programmer, Squad) | ✅ ON |

---

## Opt-Out

```yaml
# .squad/rubber-duck.yml
enabled: false
```

---

## Related PRs
- AgentCraftworks core: https://github.com/AgentCraftworks/AgentCraftworks/pull/862
- AgentCraftworks-WebSite: https://github.com/AgentCraftworks/AgentCraftworks-WebSite/pull/198

### CODE REVIEW

File: docs/RUBBER_DUCK_INTEGRATION.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+```yaml
+rubberDuck:
+  enabled: true
+  minEngagementLevel: 3   # Level 3 = Peer Programmer and above
+  triggerCheckpoints:
+    afterPlan: true
+    afterComplexImpl: true
+    afterTestWrite: true
+  maxCallsPerHour: 10
+```
```
[the remark this answers is not in the document]
Comment (Jenp-AICraftWorks) [reply docs/RUBBER_DUCK_INTEGRATION.md]: @copilot apply changes based on [this feedback](https://github.com/AgentCraftworks/AgentCraftworks-CE/pull/162#discussion_r3107413232)

File: docs/RUBBER_DUCK_INTEGRATION.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+> **Status**: CURRENT  
+> **Edition**: Community Edition (CE) — open-source (MIT)  
+> **Related**: [ENGAGEMENT_LEVELS.md (CE)](./architecture.md) · [GitHub Copilot Rubber Duck](https://github.blog/ai-and-ml/github-copilot/github-copilot-cli-combines-model-families-for-a-second-opinion/)
+
+---
+
+## What Is Rubber Duck?
+
+**GitHub Copilot Rubber Duck** (released April 6, 2026) adds a cross-model-family second opinion to your agent workflow. When your primary agent uses Claude, Rubber Duck runs on GPT-5.4 — catching what the primary agent misses at three key checkpoints:
+
```
[the remark this answers is not in the document]
Comment (Jenp-AICraftWorks) [reply docs/RUBBER_DUCK_INTEGRATION.md]: @copilot apply changes based on [this feedback](https://github.com/AgentCraftworks/AgentCraftworks-CE/pull/162#discussion_r3107413223)

**Review by jenperret — APPROVED** [review]

**Comment by Jenp-AICraftWorks** [comment]
I have read the CLA Document and I hereby sign the CLA.
